ci-operator/step-registry/workers-rhel/pre-hook/sshtunnel/workers-rhel-pre-hook-sshtunnel-commands.sh

```diff
@@ -0,0 +1,205 @@
+  gather_facts: false
+
+  vars:
+    platform_version: "{{ lookup('env', 'PLATFORM_VERSION') }}"
+    major_platform_version: "{{ platform_version[:1] }}"
+    bastion_private_address: "{{ lookup('env', 'BASTION_PRIVATE_ADDRESS') }}"
+    bastion_ssh_user: "{{ lookup('env', 'BASTION_SSH_USER') }}"
+    bastion_ssh_private_key_file: "{{ lookup('env', 'BASTION_SSH_PRIV_KEY_PATH') }}"
+
+  tasks:
+  - name: Create SSH tunnel in disconnected network env on public cloud platform
+    block:
+    - name: Copy private SSH
+      copy:
+        src: "{{ bastion_ssh_private_key_file }}"
+        dest: "/tmp/id_rsa.pem"
+        mode: '0600'
+
+    - name: Create ssh tunnel service
+      template:
+        src: qe-ssh-tunnel.service
+        dest: /etc/systemd/system/qe-ssh-tunnel.service
+
+    # Install the SeLinux policy for qe-ssh-tunnel.service on RHEL-8
+    - block:
+      - name: Copy qe-ssh-tunnel.te
+        copy:
+          src: qe-ssh-tunnel.te
+          dest: /tmp/qe-ssh-tunnel.te
+
+      - name: Convert TE file into a policy moduel
+        shell: checkmodule -M -m -o /tmp/qe-ssh-tunnel.mod /tmp/qe-ssh-tunnel.te
+
+      - name: compile and generate policy package
+        shell: semodule_package -o /tmp/qe-ssh-tunnel.pp -m /tmp/qe-ssh-tunnel.mod
+
+      - name: Loaded qe-ssh-tunnel policy package
+        shell: sudo semodule -B && sudo semodule -i /tmp/qe-ssh-tunnel.pp
+
+      - name: list qe-ssh-tunnel policy module
+        shell: sudo semodule -l | grep qe-ssh-tunnel
+        register: res
+
+      - debug: var=res.stdout
+      when: major_platform_version == "8"
+
+    - name: Set the SElinux context for id_rsa.pem
+      shell: chcon -t user_home_t /tmp/id_rsa.pem
+      when: major_platform_version == "8"
+
+    - name: Start ssh tunnel service
+      systemd:
+        name: qe-ssh-tunnel
+        enabled: yes
+        daemon_reload: yes
+        state: started
+
+    - name: Check tunnel is running
+      shell: ss -tlnup | grep 5080
+      register: test_ssh_tunnel
+      retries: 2
+      delay: 50
+      until: "test_ssh_tunnel.rc == 0"
+
+    - debug: var=test_ssh_tunnel
+
+    - set_fact:
+        socks_proxy: "socks5h://localhost:5080"
+
+    - name: Configure ssh yum
+      blockinfile:
+        path: /etc/yum.conf
+        block: |
+          proxy={{socks_proxy}}
+    when:
+      - bastion_private_address != ""
+EOF
+
+cat > qe-ssh-tunnel.service <<- EOF
+[Unit]
+Description=OpenShift QE SSH Tunnel
+After=network.target syslog.target sshd.service
+
+[Service]
+Type=simple
+TimeoutStartSec=5m
+ExecStart=/usr/bin/ssh -i /tmp/id_rsa.pem -o ConnectTimeout=60 -o ServerAliveCountMax=2 -o ServerAliveInterval=75 -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o VerifyHostKeyDNS=yes -p 22 -ND 5080 {{ bastion_ssh_user }}@{{ bastion_private_address }}
+Restart=always
+RestartSec=30
+
+[Install]
+WantedBy=multi-user.target
+EOF
+
+# Generate qe-ssh-tunnel type enforcemnet(TE) file
+cat > qe-ssh-tunnel.te <<- EOF
```

Comment:
Maybe we can convert this step to a ansible step via ansible copy module.

Reply:
changed to use copy module to write a file with content